applications/nrf_desktop/src/modules/hids.c

```diff
@@ -496,15 +496,15 @@ static void notify_hids(const struct ble_peer_event *event)
 		cur_conn = NULL;
 		secured = false;
 		if (CONFIG_DESKTOP_HIDS_FIRST_REPORT_DELAY > 0) {
-			k_delayed_work_cancel(&notify_secured);
```

Comment:
Ok. Cancel will not race with any resubmit - all operations are in sync with system workqueue.